graylog2-server/src/main/java/org/graylog2/shared/rest/documentation/generator/Generator.java

```diff
@@ -81,18 +83,42 @@
     private static Map<String, Object> overviewResult = Maps.newHashMap();
     private static Reflections reflections;
 
+    private final Set<String> packageNames;
+    private final Map<String, String> pluginMapping;
+    private final String pluginPathPrefix;
     private final ObjectMapper mapper;
 
-    public Generator(String[] packageNames, ObjectMapper mapper) {
+    public Generator(Set<String> packageNames, Map<String, String> pluginMapping, String pluginPathPrefix, ObjectMapper mapper) {
+        this.packageNames = packageNames;
+        this.pluginMapping = pluginMapping;
+        this.pluginPathPrefix = pluginPathPrefix;
         this.mapper = mapper;
 
         if (reflections == null) {
-            reflections = new Reflections((Object[]) packageNames);
+            reflections = new Reflections((Object[]) packageNames.toArray());
         }
     }
 
     public Generator(String packageName, ObjectMapper mapper) {
-        this(new String[]{packageName}, mapper);
+        this(ImmutableSet.of(packageName), ImmutableMap.of(), "", mapper);
+    }
+
+    private String prefixedPath(Class<?> resourceClass, @Nullable String resourceAnnotationPath) {
+        final String resourceClassName = resourceClass.getCanonicalName();
+        final String resourcePath = (resourceAnnotationPath == null ? "" : resourceAnnotationPath);
```

Comment:
Minor nitpick: This could be replaced by [`Strings#nullToEmpty(String)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Strings.html#nullToEmpty(java.lang.String)).